src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -37,11 +42,14 @@ class LeonardoService(gdDAO: DataprocDAO, dbRef: DbReference)(implicit val execu
 
   def deleteCluster(googleProject: GoogleProject, clusterName: String): Future[Int] = {
     getClusterDetails(googleProject, clusterName) flatMap  { cluster =>
-      if(cluster.status.isActive) {
-        for {
-          _ <- gdDAO.deleteCluster(googleProject, clusterName)
-          recordCount <- dbRef.inTransaction(dataAccess => dataAccess.clusterQuery.deleteCluster(cluster.googleId))
-        } yield recordCount
+      if (cluster.status.isActive) {
```

Comment:
A cluster status of `Error` is not part of the set of active statuses, this means we cannot delete clusters with a status of `Error`. Is this the behavior we want?

Reply:
Good catch. Per Hussein's earlier comment, I'm going to make the ClusterMonitorActor auto-delete clusters in Error state. So maybe it makes sense to keep this logic, to prevent the user from messing with the ClusterMonitorActor flow (i.e. we don't want users deleting Error'd clusters if they're already being managed by the ClusterMonitorActor).

Reply:
Then again, I'm only setting status to `Error` for non-recoverable errors -- so we do probably want to allow the user to delete in that case. I'll fix this -- thanks!